### PR TITLE
Retrying: Make the testing scripts executable on Python 3.x

### PR DESCRIPTION
**Pull request by thunderboltsid**
Fixing the messup in [PR#7113](https://github.com/grpc/grpc/pull/7113)

### CODE REVIEW

**Comment by kpayson64** [comment]
I don't have much context on this change yet, what is the generate_projects.sh changes for?

**Comment by thunderboltsid** [comment]
@kpayson64 so, six needed to be added to all dockerfiles which requires modifying the dockerfile templates and then generating the dockerfiles and other assets using generate_projects.sh

**Comment by kpayson64** [comment]
Would it work to just add the templates/ and tools/ directories to this commit?  I'm not really equipped to evaluate the -25,000 lines from the generate_projects.sh run.

**Comment by thunderboltsid** [comment]
@kpayson64 It shouldn't be ~25k lines anymore. More like 500. Take a look.

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -952,18 +952,18 @@ def git(cmd, cwd=cwd):
...
 
 if args.use_docker:
   if not args.travis:
-    print 'Seen --use_docker flag, will run tests under docker.'
-    print
-    print 'IMPORTANT: The changes you are testing need to be locally committed'
-    print 'because only the committed changes in the current branch will be'
-    print 'copied to the docker environment.'
+    print('Seen --use_docker flag, will run tests under docker.')
+    print()
```
Comment (kpayson64) [inline tools/run_tests/run_tests.py]: print() -> print('')

**Comment by kpayson64** [comment]
LGTM after the comments.

**Comment by thunderboltsid** [comment]
@kpayson64 made changes accordingly and added the import for print_function from future in files that used the print statement

**Comment by kpayson64** [comment]
@thunderboltsid 
The `__future__` import statements should appear in their own block before all other imports.

**Comment by thunderboltsid** [comment]
@kpayson64 Fixed. :)

**Comment by kpayson64** [comment]
LGTM

**Comment by thunderboltsid** [comment]
@nicolasnoble It'd be great if you could take a look at this as well

**Pull request merged**